service/src/main/resources/api/service_openapi.yaml

```diff
@@ -2259,6 +2304,86 @@ components:
           format: uuid
         dataset:
           $ref: "#/components/schemas/GcpBigQueryDatasetResource"
+    CloneWorkspaceRequest:
+      description: >-
+        Request body for cloning an entire workspace. The stage is always MC. Cloning instructions
+        are taken from individual resources.
+      type: object
+      required:
+        - spendProfile
+      properties:
+        displayName:
+          description: The human readable name of the workspace
+          type: string
+        description:
+          description: A description of the cloned workspace
+          type: string
+        spendProfile:
+          description: ID of provided spend profile
+          type: string
+        location:
+          description: >-
+            GCP Location to use for cloud-based resources. If omitted, the location of the source
+            resource will be used.
+          type: string
+    CloneWorkspaceResult:
+      description: >-
+        Response for workspace clone, including details on each controlled or referenced resource
+        either cloned or attempted to be cloned.
+      type: object
+      properties:
+        jobReport:
+          $ref: '#/components/schemas/JobReport'
+        errorReport:
+          $ref: '#/components/schemas/ErrorReport'
+        workspace:
+          $ref: '#/components/schemas/ClonedWorkspace'
+    ClonedWorkspace:
+      description: >-
+        Flight result class for workspace clone. Indicates status for each resource
+        cloned (or skipped).
+      type: object
+      properties:
+        sourceWorkspaceId:
+          type: string
+          format: uuid
+        destinationWorkspaceId:
+          type: string
+          format: uuid
+        resources:
+          type: array
+          items:
+            $ref: '#/components/schemas/ResourceCloneDetails'
+    ResourceCloneDetails:
+      description: >-
+        A structure to give provenance of a controlled or referenced resource clone
+      type: object
+      properties:
+        cloningInstructions:
+          $ref: '#/components/schemas/CloningInstructionsEnum'
+        resourceType:
+          $ref: '#/components/schemas/ResourceType'
+        stewardshipType:
+          $ref: '#/components/schemas/StewardshipType'
+        sourceResourceId:
+          description: resource ID of the original resource
+          type: string
+          format: uuid
+        destinationResourceId:
+          description: resource ID of the cloned resource, or null if failed or skipped
+          type: string
+          format: uuid
+        result:
+          $ref: '#/components/schemas/CloneResourceResult'
+        errorMessage:
+          description: If result is FAILED, a short description of the cause, if available.
+          type: string
+    CloneResourceResult:
```

Comment:
After talking with @marikomedlock, we decided we also need the resource name and description here. In the CLI, the names are more often used than the IDs in order to address resources.

Reply:
I think the resource name + description should be included in the full workspace clone (or have an option to do so). One way the solutions team plans to use MC Terra and the CLI is to write scripts/workflows that refer to workspace resources by their name (unique within a workspace). Then when they clone the workspace, the same scripts/workflows should still work even though the underlying cloud resources are different. This depends on the resource names being the same.

I would also like to include the resource name + description in the CLI command output for `terra workspace clone`. This doesn't have to mean modifying the WSM API per se, though that is one way to implement it. The CLI could also e.g. make a separate request to fetch the resource name + descriptions.